### PR TITLE
Support for after_call state when not returning agents to the queue automatically

### PR DESCRIPTION
**Pull request by benlangfeld**

### CODE REVIEW

File: lib/electric_slide/call_queue.rb [hunk 1]
```diff
@@ -219,6 +219,7 @@ def conditionally_return_agent(agent, return_method = @agent_return_method)
         return_agent agent
       else
         logger.debug "Not returning agent #{agent.inspect} to the queue"
+        return_agent agent, :after_call
```
Comment (bklang) [inline lib/electric_slide/call_queue.rb]: This change introduces a new state to ElectricSlide, `:after_call`. It needs to be documented on the `Agent` class and probably the README.  Also, we need to think through whether we want to offer a "mode" to ElectricSlide consumers where agents check themselves back in as `:available` immediately, or wait for manual intervention.

Comment (benlangfeld) [inline lib/electric_slide/call_queue.rb]: That "mode" is https://github.com/adhearsion/electric_slide/blob/develop/lib/electric_slide/call_queue.rb#L25-L28

Comment (bklang) [inline lib/electric_slide/call_queue.rb]: Also a new state should have a corresponding callback (see #26)

Comment (benlangfeld) [inline lib/electric_slide/call_queue.rb]: There is nothing in the README about agent states right now. I updated what API documentation there is on Agent. Any thoughts on renaming `:busy` to `:on_call` to match this?

Comment (bklang) [inline lib/electric_slide/call_queue.rb]: > Any thoughts on renaming :busy to :on_call to match this?

:+1+

Comment (bklang) [inline lib/electric_slide/call_queue.rb]: Should `:available` also become `:waiting_for_call`?

Comment (benlangfeld) [inline lib/electric_slide/call_queue.rb]: I think `:available` is fine.

**Pull request merged**